src/main/scala/com/gu/certificate/Magic.scala

```diff
@@ -87,13 +89,22 @@ object Magic extends BouncyCastle with FileHelpers {
       getChainFromCertificate(certificate).map(toPem(_).trim).mkString("\n")
     }
 
-    System.err.println(s"installing to IAM")
-
     val pathPrefix = path.getOrElse("").stripPrefix("/").replace("/", "-")
+    val certificateName = s"$pathPrefix$safeDomain-exp$expDate"
+
+    service match {
+      case Some("iam") => installToIAM(path, region, installCredentialsProvider, certificateName, decryptedPem, certificatePem, chainPem)
+      case Some("apigateway") => installToGateway(region, installCredentialsProvider, safeDomain, certificateName, decryptedPem, certificatePem, chainPem)
+      case _ => None
```

Comment:
this is a bit weird, but at least `case _` is covered by `scopt` input validation.